File: internal/app/user_service.go
package app

import (
	"context"
	"spahtmx/internal/domain"
)

type UserService struct {
	repo domain.UserRepository
}

func NewUserService(r domain.UserRepository) *UserService {
	return &UserService{
		repo: r,
	}
}

func (s *UserService) GetUsers(ctx context.Context) ([]domain.User, error) {
	return s.repo.GetUsers(ctx)
}

func (s *UserService) UpdateUserStatus(ctx context.Context, id string) error {
	if id == "" {
		return domain.ErrInvalidInput
	}
	return s.repo.UpdateUserStatus(ctx, id)
}

func (s *UserService) GetUserCount(ctx context.Context) string {
	return "1234"
}

func (s *UserService) GetPageView(ctx context.Context) string {
	return "1212121"
}

File: src/app/user_service.rs
use std::sync::Arc;
use crate::domain::repositories::UserRepository;
use crate::domain::models::User;

pub struct UserService {
    repo: Arc<dyn UserRepository>,
}

impl UserService {
    pub fn new(repo: Arc<dyn UserRepository>) -> Self {
        Self { repo }
    }

    pub async fn get_users(&self) -> Result<Vec<User>, Box<dyn std::error::Error + Send + Sync>> {
        self.repo.get_users().await
    }

    pub async fn update_user_status(&self, id: &str) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
        if id.is_empty() {
            return Err("Invalid input".into());
        }
        let mut u = self.repo.get_user(id).await?;
        u.status = !u.status;
        self.repo.update_user(u).await
    }

    pub fn get_user_count(&self) -> String {
        "1234".to_string()
    }

    pub fn get_page_view(&self) -> String {
        "1212121".to_string()
    }
}
